File: rest/model/serviceEndpoint.js
// SPDX-License-Identifier: Apache-2.0

class ServiceEndpoint {
  static DOMAIN_NAME = 'domain_name';
  static IP_ADDRESS_V4 = `ip_address_v4`;
  static PORT = 'port';

  constructor(serviceEndpoint) {
    this.domainName = serviceEndpoint.domain_name;
    this.ipAddressV4 = serviceEndpoint.ip_address_v4;
    this.port = serviceEndpoint.port;
  }
}

export default ServiceEndpoint;

File: common/src/main/java/org/hiero/mirror/common/domain/node/ServiceEndpoint.java
// SPDX-License-Identifier: Apache-2.0

package org.hiero.mirror.common.domain.node;

import lombok.Builder;

@Builder(toBuilder = true)
public record ServiceEndpoint(String domainName, String ipAddressV4, int port) {
    public static final ServiceEndpoint CLEAR = new ServiceEndpoint(null, null, -1);
}
